### ISSUE TITLE
the error on training with VeRi

### ISSUE THREAD

**Issue by MingjieLiu** [nwojke/cosine_metric_learning#8]
Hi, @nwojke 
thank you for your sharing the training code of deep_sort

I train it on VeRi dataset, and use the code you provide for it. How it shows some error, I google it, but there is no answer like that kind of problem
![image](https://user-images.githubusercontent.com/16024855/40286442-ecc114a6-5ce1-11e8-9866-15187aba5a33.png)
![image](https://user-images.githubusercontent.com/16024855/40286450-fca95cca-5ce1-11e8-9246-8f79354be6eb.png)
![image](https://user-images.githubusercontent.com/16024855/40286478-28bf9cac-5ce2-11e8-8ac4-3938d81b2360.png)

Could you help me? thank you.

**Comment by nwojke** [comment]
From a first glance it looks like something is wrong with the input image shape.. I will have a look and re-test the code but it will take some time, sorry.

**Comment by nwojke** [comment]
I was able to reproduce the error. As a work-around, I changed the code to load all images into memory instead of loading them during preprocessing. It consumes more memory but works.

**Comment by sherlockchou86** [comment]
good, it works!

**Comment by nwojke** [comment]
Glad to hear. Sorry that it took so long.

**Comment by srika91** [comment]
@sherlockchou86 what is the best validation accuracy you are getting on the vehicle reidentification dataset? I am not able to reduce the loss below 1.9 and the best validation accuracy i got was 68%

**Comment by irisqp** [comment]
> 
> 
> Hi, @nwojke
> thank you for your sharing the training code of deep_sort
> 
> I train it on VeRi dataset, and use the code you provide for it. How it shows some error, I google it, but there is no answer like that kind of problem
> ![image](https://user-images.githubusercontent.com/16024855/40286442-ecc114a6-5ce1-11e8-9866-15187aba5a33.png)
> ![image](https://user-images.githubusercontent.com/16024855/40286450-fca95cca-5ce1-11e8-9246-8f79354be6eb.png)
> ![image](https://user-images.githubusercontent.com/16024855/40286478-28bf9cac-5ce2-11e8-8ac4-3938d81b2360.png)
> 
> Could you help me? thank you.

hello，Thanks for your work. I want to train it on VeRi DataSet too, but I can't get data from the original owner, I emailed but no one reply me. Would you please help me? Send me a copy of dataset is okay, or would you share me with your trained results like .pb file. thank you.

**Comment by irisqp** [comment]
> 
> 
> Hi, @nwojke
> thank you for your sharing the training code of deep_sort
> 
> I train it on VeRi dataset, and use the code you provide for it. How it shows some error, I google it, but there is no answer like that kind of problem
> ![image](https://user-images.githubusercontent.com/16024855/40286442-ecc114a6-5ce1-11e8-9866-15187aba5a33.png)
> ![image](https://user-images.githubusercontent.com/16024855/40286450-fca95cca-5ce1-11e8-9246-8f79354be6eb.png)
> ![image](https://user-images.githubusercontent.com/16024855/40286478-28bf9cac-5ce2-11e8-8ac4-3938d81b2360.png)
> 
> Could you help me? thank you.

你好，我也遇到了和你差不多一样的问题，请问你当时是怎么解决的？

**Comment by zengdiqing1994** [comment]
> > Hi, @nwojke
> > thank you for your sharing the training code of deep_sort
> > I train it on VeRi dataset, and use the code you provide for it. How it shows some error, I google it, but there is no answer like that kind of problem
> > ![image](https://user-images.githubusercontent.com/16024855/40286442-ecc114a6-5ce1-11e8-9866-15187aba5a33.png)
> > ![image](https://user-images.githubusercontent.com/16024855/40286450-fca95cca-5ce1-11e8-9246-8f79354be6eb.png)
> > ![image](https://user-images.githubusercontent.com/16024855/40286478-28bf9cac-5ce2-11e8-8ac4-3938d81b2360.png)
> > Could you help me? thank you.
> 
> 你好，我也遇到了和你差不多一样的问题，请问你当时是怎么解决的？

老哥，我也发了邮件给veri数据的作者，也没有回复，你现在已经拿到了文件训练了吗？可以加个联系方式交流一下吗？我的微信zengdiqing94，万分感激！！！

**Comment by JIALIUZHOU** [comment]
> > > Hi, @nwojke
> > > thank you for your sharing the training code of deep_sort
> > > I train it on VeRi dataset, and use the code you provide for it. How it shows some error, I google it, but there is no answer like that kind of problem
> > > ![image](https://user-images.githubusercontent.com/16024855/40286442-ecc114a6-5ce1-11e8-9866-15187aba5a33.png)
> > > ![image](https://user-images.githubusercontent.com/16024855/40286450-fca95cca-5ce1-11e8-9246-8f79354be6eb.png)
> > > ![image](https://user-images.githubusercontent.com/16024855/40286478-28bf9cac-5ce2-11e8-8ac4-3938d81b2360.png)
> > > Could you help me? thank you.
> > 
> > 
> > 你好，我也遇到了和你差不多一样的问题，请问你当时是怎么解决的？
> 
> 老哥，我也发了邮件给veri数据的作者，也没有回复，你现在已经拿到了文件训练了吗？可以加个联系方式交流一下吗？我的微信zengdiqing94，万分感激！！！

我之前训练也遇到这个问题，后来用整个的作者的代码就没有问题了
